src/main/com/aca/homework/week7/clone/Light.java

```diff
@@ -0,0 +1,39 @@
+package com.aca.homework.week7.clone;
+
+public class Light implements Cloneable{
+    private State state;
+    private static Light[] lights = new Light[2];
+    private Light(boolean state){
+        this.state = state ? State.ON : State.OFF;
+    }
+
+    public static Light lightOf(boolean state){
+        int index = getIndexByState(state);
+        if(objectExists(state)){
```

Comment:
no need to cache the Ligh instances. The requirement was to create a static factory only. also, please create separate methods for off and on state light